src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/ReportsService.java

```diff
@@ -76,7 +76,18 @@ public ReportsService(
             .stream()
             .map(this::fromDbZipfileSummary)
             .filter(summary -> isEmpty(container) || summary.container.equalsIgnoreCase(container))
-            .filter(summary -> classification == null || summary.classification.equalsIgnoreCase(classification.name()))
+            .filter(summary -> {
+                try {
+                    if (classification == null) {
+                        return true;
+                    }
+
+                    return summary.classification.equalsIgnoreCase(classification.name());
```

Comment:
if you compare like this you will not get NPE
```suggestion
                    return classification.name.equalsIgnoreCase(summary.classification);
```

Reply:
I think this is cleaner for method
```suggestion
                          var stream=  zipFilesSummaryRepository.getZipFileSummaryReportFor(date)
            .stream()
            .map(this::fromDbZipfileSummary)
            .filter(summary -> isEmpty(container) || summary.container.equalsIgnoreCase(container));

        if (classification != null) {
            stream.filter(summary -> classification.name().equalsIgnoreCase(summary.classification));
        }
        return stream.collect(Collectors.toList());
```